auth/README.md

```diff
@@ -0,0 +1,6 @@
+# About the model
+
+We use per-resource-type permissions off of "project" because
+we don't allow granting permissions on individual resources, only
+on projects.  This allows us to minimize the amount of state we
```

Comment:
(I know I was arguing in favour of only importing the projects yesterday, too)

If we ever decide to make the model more granular to make a distinction between "a user can delete a repo" and "a user can delete a build environment", apart from updating the rules, how disruptive would changing of the state we keep in the DB be? Does FGA allow "migrations" of its DB?

Reply:
It is possible and there is a process for it https://openfga.dev/docs/modeling/migrating/migrating-relations

Reply:
I'm imagining that we'd define the new relations in the model, possibly bootstrapping off the existing permissions at first as:

```
type rule_type
  relations
    define project: [project]

    define getter: [user] or rule_type_getter from project
    define creator: [user] or rule_type_creator from project
    define updater: [user] or rule_type_updater from project
    define deleter: [user] or rule_type_deleter from project
```

Next, we would update our code on create / update of resources to create the relationships as resources were updated.  Following that, we would run a big pass to sync all our existing objects into the relationship database. (Basically, this is a dual-write scheme -- we might end up syncing some relationships _twice_, but we wouldn't miss any if we start with the live updates.)

Once we think the database is synced, we could do a "dark launch" where we check both the existing project-level permission _and_ the new permission and log any differences, using the result from the legacy permission.  Assuming no differences during the launch window, we'd switch the code to checking only the new permission, then remove the `resource_actioner` permissions from project.